### PR TITLE
feat: add Kafka SASL support

### LINKED ISSUE #6889: Support Kafka SCRAM auth
https://github.com/PostHog/posthog/pull/3280

### PR DESCRIPTION
**Pull request by joetrollo**
## Problem

Confluent Cloud (one of the recommended Kafka providers in the PostHog docs) requires SASL authentication to connect, which requires passing additional configuration to the underlying Kafka libraries.

## Changes

This PR adds new environment variables to configure the security protocol and the SASL mechanism, username, and password, and has both Python and TS code pass them along when appropriate.

## How did you test this code?

I built a Docker image with these changes and deployed them to my test cluster with Confluent Cloud. Events were successfully ingested and ended up in ClickHouse.

Closes #6889

### CODE REVIEW

**Comment by guidoiaquinti** [comment]
Looping @yakkomajuri and @tiina303 from the ingestion team

**Review by tiina303 — COMMENTED** [review]
Looks fine to me, adding a test could be nice, will kick off the tests.

File: ee/kafka_client/test/test_client.py [hunk 1]
```diff
@@ -28,3 +31,42 @@ def test_kafka_produce_and_consume(self):
         producer.produce(topic=self.topic, data=self.payload)
         payload = next(consumer)
         self.assertEqual(payload.value, self.payload)
+
+    def test_kafka_default_security_protocol(self):
+        producer = _KafkaProducer(test=False)
+        self.assertEqual(producer.producer.config["security_protocol"], "PLAINTEXT")
```
Comment (tiina303) [inline ee/kafka_client/test/test_client.py:37]: Looks like this test is failing, from the [test run](https://github.com/PostHog/posthog/runs/6006002876):
```
ee/kafka_client/test/test_client.py:37: error: "TestKafkaProducer" has no attribute "config"
ee/kafka_client/test/test_client.py:56: error: "TestKafkaProducer" has no attribute "config"
ee/kafka_client/test/test_client.py:72: error: "TestKafkaProducer" has no attribute "config"
```

File: posthog/settings/data_stores.py [hunk 2]
```diff
@@ -115,6 +115,11 @@
 
 KAFKA_BASE64_KEYS = get_from_env("KAFKA_BASE64_KEYS", False, type_cast=str_to_bool)
 
+KAFKA_SECURITY_PROTOCOL = os.getenv("KAFKA_SECURITY_PROTOCOL")
```
Comment (yakkomajuri) [inline posthog/settings/data_stores.py:118]: let's use explicit defaults here, even if it's `None`. That's how things are being done elsewhere in this file.

File: plugin-server/src/utils/db/hub.ts [hunk 3]
```diff
@@ -107,6 +107,17 @@ export async function createHub(
             on the ca trust_cert for this purpose. */
             rejectUnauthorized: false,
         }
+    } else if (serverConfig.KAFKA_SECURITY_PROTOCOL === 'SSL' || serverConfig.KAFKA_SECURITY_PROTOCOL === 'SASL_SSL') {
```
Comment (yakkomajuri) [inline plugin-server/src/utils/db/hub.ts:110]: nit: can we set up an enum for these?

File: plugin-server/src/types.ts [hunk 4]
```diff
@@ -66,6 +66,10 @@ export interface PluginsServerConfig extends Record<string, any> {
     KAFKA_CLIENT_CERT_B64: string | null
     KAFKA_CLIENT_CERT_KEY_B64: string | null
     KAFKA_TRUSTED_CERT_B64: string | null
+    KAFKA_SECURITY_PROTOCOL: string | null
+    KAFKA_SASL_MECHANISM: 'plain' | 'scram-sha-256' | 'scram-sha-512' | null
```
Comment (yakkomajuri) [inline plugin-server/src/types.ts:70]: nit: let's maybe use an enum?

**Review by yakkomajuri — COMMENTED** [review]
a few nits from my side, else looks good, thanks for this!

File: plugin-server/src/utils/db/hub.ts (hunk 3 above)
Comment (joetrollo) [reply plugin-server/src/utils/db/hub.ts:110]: Sure, would you also prefer an enum for the values in Python?

Comment (yakkomajuri) [reply plugin-server/src/utils/db/hub.ts:110]: Could be nice yeah, although I'm being more particular about TS as we use them across the board. Unsure how consistent we are about this on the Python side.

**Comment by joetrollo** [comment]
Hi all, is there any further feedback for me to implement on this PR?

**Review by fuziontech — APPROVED** [review]
Fantastic! Thank you so much for submitting this PR @joetrollo!

**Pull request merged**